src/Codeception/PHPUnit/Listener.php

```diff
@@ -10,119 +11,246 @@
 use PHPUnit_Framework_Test;
 use Symfony\Component\EventDispatcher\EventDispatcher;
 
-class Listener implements \PHPUnit_Framework_TestListener
-{
-    /**
-     * @var \Symfony\Component\EventDispatcher\EventDispatcher
-     */
-    protected $dispatcher;
+// PHPUnit 6 definition for the TestListener class includes a new method `addWarning`
+if (interface_exists('PHPUnit\Framework\TestListener')) {
```

Comment:
It is an overkill to provide 2 versions of the class, simply add `addWarning` method, it doesn't matter if it is missing in the earlier versions of the interface.